src/video_stream.cpp

```diff
@@ -241,16 +243,22 @@ virtual void subscribe() {
     cap->open(video_stream_provider);
     if(video_stream_provider_type == "videofile" )
       {
-        if(latest_config.stop_frame == -1) latest_config.stop_frame = cap->get(cv::CAP_PROP_FRAME_COUNT);
+        // We can only check the number of frames when we actually open the video file
+        NODELET_INFO_STREAM("Video number of frames: " << cap->get(cv::CAP_PROP_FRAME_COUNT));
+        if(latest_config.stop_frame == -1)
+        {
+          NODELET_WARN_STREAM("'stop_frame' set to -1, setting internally (won't be shown in dynamic_reconfigure) to last frame: " << cap->get(cv::CAP_PROP_FRAME_COUNT));
```

Comment:
Try to avoid doing things obscurely and warn the user. The user may want to use this information to update their launch file to the last frame.